### PR TITLE
[GraphBolt] convert item list to MiniBatch

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

Add mapper to instantiate `MiniBatch`. Removals of `minibatch_converter` in other testcases will be done in another PR after this one is merged.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/item_sampler.py [hunk 1]
```diff
@@ -2,16 +2,53 @@
...
 from ..batch import batch as dgl_batch
 from ..heterograph import DGLGraph
 from .itemset import ItemSet, ItemSetDict
+from .minibatch import MiniBatch
+
+__all__ = ["ItemSampler", "minibatcher_default"]
+
 
-__all__ = ["ItemSampler"]
+def minibatcher_default(batch, names):
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:20]: What's `names`?

Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:20]: names of item in `batch` such as `seed_nodes`, `node_pairs` which correspond to data attributes of `MiniBatch`.

Comment (frozenbugs) [reply python/dgl/graphbolt/item_sampler.py:20]: add doc string?

Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:20]: added.

File: python/dgl/graphbolt/item_sampler.py [hunk 2]
```diff
@@ -2,16 +2,61 @@
...
+        A minibatch.
+    """
+    if names is None:
+        dgl_warning(
+            "Failed to map item list to `MiniBatch` as the names of items are "
+            "not provided. Please provide a customized `MiniBatcher`. "
+            "The item list is returned as is."
+        )
+        return batch
+    if len(names) == 1:
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:48]: Add a comment to elaborate why we need to have a special handle for `len(names) == 1`.

File: python/dgl/graphbolt/item_sampler.py [hunk 3]
```diff
@@ -2,16 +2,61 @@
...
+    same names as the items. The names of items are supposed to be provided
+    and align with the data attributes of `MiniBatch`. If any unknown item name
+    is provided, exception will be raised. If the names of items are not
+    provided, the item list is returned as is and a warning will be raised.
+
+    Parameters
+    ----------
+    batch : list
+        List of items.
+    names : Tuple[str] or None
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:33]: does the len of names should match the len of batch?

File: python/dgl/graphbolt/item_sampler.py [hunk 4]
```diff
@@ -2,16 +2,63 @@
...
+    if names is None:
+        dgl_warning(
+            "Failed to map item list to `MiniBatch` as the names of items are "
+            "not provided. Please provide a customized `MiniBatcher`. "
+            "The item list is returned as is."
+        )
+        return batch
+    if len(names) == 1:
+        # Handle the case of single item: batch = tensor([0, 1, 2, 3]), names =
+        # ("seed_nodes",).
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:50]: It is obvious that this code handles the case of single item, the comment should focus on why single item needs to have a special handle.

File: python/dgl/graphbolt/item_sampler.py [hunk 5]
```diff
@@ -2,16 +2,63 @@
...
+__all__ = ["ItemSampler", "minibatcher_default"]
+
+
+def minibatcher_default(batch, names):
+    """Default minibatcher.
+
+    The default minibatcher maps a list of items to a `MiniBatch` with the
+    same names as the items. The names of items are supposed to be provided
+    and align with the data attributes of `MiniBatch`. If any unknown item name
+    is provided, exception will be raised. If the names of items are not
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:26]: Other than raise an exception, can we blindly hold the data in minibatch?

Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:26]: yes, we can. But I don't think it's a good behavior to add data attributes into `MiniBatch` though it's quite easy to achieve in python via `setattr()`

File: python/dgl/graphbolt/item_sampler.py (hunk 3 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:33]: yes, this should have been verified in instantiating of `ItemSet`.

File: python/dgl/graphbolt/item_sampler.py (hunk 2 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:48]: added

File: python/dgl/graphbolt/item_sampler.py (hunk 3 above)
Comment (frozenbugs) [reply python/dgl/graphbolt/item_sampler.py:33]: Add doc string here should be good.

File: python/dgl/graphbolt/item_sampler.py (hunk 5 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:26]: as discussed offline, let's support such a bit **wild** behaviour

File: python/dgl/graphbolt/item_sampler.py (hunk 3 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py:33]: added

File: python/dgl/graphbolt/item_sampler.py (hunk 4 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/item_sampler.py]: reason detail is added

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**